registry/registry.go

```diff
@@ -444,12 +446,55 @@ func createVersion(c *Space, db *kivik.DB, ver *Version, attachments []*kivik.At
 	return nil
 }
 
-func CreatePendingVersion(c *Space, ver *Version, attachments []*kivik.Attachment, app *App, editor *auth.Editor) (err error) {
-	return createVersion(c, c.PendingVersDB(), ver, attachments, app, editor)
+func CreatePendingVersion(c *Space, ver *Version, attachments []*kivik.Attachment, app *App) error {
+	return createVersion(c, c.PendingVersDB(), ver, attachments, app, true)
+}
+
+func CreateReleaseVersion(c *Space, ver *Version, attachments []*kivik.Attachment, app *App) error {
+	// We need to skip version check, because we don't drop pending
+	// version until the end to avoid data loss in case of error
+	return createVersion(c, c.VersDB(), ver, attachments, app, false)
```

Comment:
In the case of an editor with `auto_publication` on, we should have the `ensureVersion` flag enabled: https://github.com/cozy/cozy-apps-registry/blob/master/router.go#L120.

The PR should be rebased on master for that.